testsuites/syncgateway/system/test_system_test.py

```diff
@@ -1,102 +1,743 @@
+    log_info('> server_seed_docs          = {}'.format(server_seed_docs))
+    log_info('> max_docs                  = {}'.format(max_docs))
+    log_info('> num_users                 = {}'.format(num_users))
+    log_info('> create_batch_size         = {}'.format(create_batch_size))
+    log_info('> create_delay              = {}'.format(create_delay))
+    log_info('> update_batch_size         = {}'.format(update_batch_size))
+    log_info('> update_delay              = {}'.format(update_delay))
+    log_info('> update_runtime_sec        = {}'.format(update_runtime_sec))
+    log_info('> changes_delay             = {}'.format(changes_delay))
+    log_info('> changes_limit             = {}'.format(changes_limit))
+    log_info('> changes_terminator_doc_id = {}'.format(changes_terminator_doc_id))
+
+    # Validate
+
+    # Server docs should be a multiple of 1000 for batching purposes
+    if server_seed_docs % 1000 != 0:
+        raise ValueError('server_seed_docs must be divisible by 1000')
+
+    # Number of docs should be equally divisible by number of users
+    if max_docs % num_users != 0:
+        raise ValueError('max_docs must be divisible by number_of_users')
+
+    # Number of docs per user (max_docs / num_users) should be equally
+    # divisible by the batch size for easier computation
+    docs_per_user = max_docs / num_users
+    if docs_per_user % create_batch_size != 0:
+        raise ValueError('docs_per_user ({}) must be devisible by create_batch_size ({})'.format(
+            docs_per_user,
+            create_batch_size
+        ))
+
+    # We want an even distributed of users per type
+    if num_users % len(USER_TYPES) != 0:
+        raise ValueError("'num_users' should be a multiple of 4")
+
+    # Make sure that the 'update_batch_size' is complatible with
+    # then number of users per type
+    num_users_per_type = num_users / len(USER_TYPES)
+    if update_batch_size > num_users_per_type:
+        raise ValueError("'batch_size' cannot be larger than number of users per type")
+
+    if num_users_per_type % update_batch_size != 0:
+        raise ValueError("'update_batch_size' ({}) should be a multiple of number_users_per_type ({})".format(
+            update_batch_size,
+            num_users_per_type
+        ))
+
+    sg_conf_name = 'sync_gateway_default'
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    # Reset cluster state
+    c = Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_conf)
 
     cluster_helper = ClusterKeywords()
     topology = cluster_helper.get_cluster_topology(cluster_config)
 
-    cbs_url = topology["couchbase_servers"][0]
-    cbs_admin_url = cbs_url.replace("8091", "8092")
+    cbs_url = topology['couchbase_servers'][0]
+    cbs_admin_url = cbs_url.replace('8091', '8092')
     cb_server = couchbaseserver.CouchbaseServer(cbs_url)
-    bucket_name = cb_server.get_bucket_names()[0]
-    cbs_ip = cb_server.host
+    bucket_name = "data-bucket"
```

Comment:
We want to ensure we are always hitting the data-bucket. This may have `cb_server.get_bucket_names()` may have different behavior depending on cc vs. di modes